### PR TITLE
Delete review comment if belongs to linting bot and outdated

### PR DESCRIPTION
**Pull request by midniteio**
@guykisel I'm actually not sure the mechanics of setting this up to test locally, but I did install github3 and test this function out outside of inline-plz and it works as expected. Thoughts?

### CODE REVIEW

**Comment by guykisel** [comment]
@midniteio currently, if you fork the repo, you need to update the travis.yml encrypted github token to be your own token if you want it to run on your fork. A crappy but working way to test this is to create a branch with intentionally bad code, then run inline-plz locally pointed at that branch's PR in order to generate comments and such.

File: inlineplz/main.py [hunk 1]
```diff
@@ -112,6 +112,8 @@ def inline(args):
         )
         if my_interface.post_messages(messages, args.max_comments) and not args.zero_exit:
             return 1
+
+        my_interface.clear_outdated_messages()
```
Comment (guykisel) [inline inlineplz/main.py]: if we're requiring this as part of the interface contract, might want to add it to the base interface implementation.

File: inlineplz/interfaces/github.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ def __init__(self, owner, repo, pr, token, url=None):
             self.github = github3.GitHub(token=token)
         else:
             self.github = github3.GitHubEnterprise(url, token=token)
+        self.user = self.github.me()
```
Comment (guykisel) [inline inlineplz/interfaces/github.py]: this seems dangerous because if you reuse your own personal account for running inline-plz, we could end up deleting legit non-inline-plz comments. maybe add some sort of inline-plz token to every comment that we can then scan for?

File: inlineplz/main.py (hunk 1 above)
Comment (guykisel) [inline inlineplz/main.py]: also note that it's possible for us to return before ever reaching this call.

Comment (midniteio) [inline inlineplz/main.py]: Yeah, I assumed if its possible that we are hitting a reason which we would exit non-zero, and posting messages had a problem for the current commit, then clearing the outdated messages might not be what we want to do.

File: inlineplz/interfaces/github.py (hunk 2 above)
Comment (midniteio) [inline inlineplz/interfaces/github.py]: Yeah that's probably much more sane. Prefix inline-plz comments, search for comments with that prefix.

File: inlineplz/main.py (hunk 1 above)
Comment (guykisel) [inline inlineplz/main.py]: makes sense.

**Comment by midniteio** [comment]
@guykisel Added to the base interface. Comments are not deleted if it is outdated and contains the prefix provided. Prefix comes from the --prefix arg, if none provided I have it set to use "## Lint Error: " as a placeholder until I get your thoughts on what might be more appropriate. It probably will need to use something as a default though, so to not remove non-inline-plz comments as you mentioned.

FYI, have not been able to run this locally yet, so treat as untested and just looking to garner feedback.

**Comment by guykisel** [comment]
so it ends up looking like this?

## Lint Error: `blah blah`

**Comment by guykisel** [comment]
what are your thoughts on having this behavior as optional vs having it as always-on?

**Comment by midniteio** [comment]
It would end up looking like:
```
## Lint Error:
{ Your lint error message(s) }
```

As for optional behavior, yeah we can wrap that around a flag for `--remove-outdated`, and if that flag is true it must also require a `--prefix flag` too then. That way you'd have control over prefix (or lack thereof, would not have to force a default then if removal is optional) independently, and also whether you want to delete outdated. Sound reasonable?

File: inlineplz/interfaces/github.py [hunk 3]
```diff
@@ -67,12 +68,12 @@ def is_duplicate(self, message, position):
...
 
     @staticmethod
-    def format_message(message):
+    def format_message(message, prefix):
         if not message.comments:
             return ''
         if len(message.comments) > 1:
             return (
-                '```\n' +
+                '```\n' + prefix + '\n' +
```
Comment (guykisel) [inline inlineplz/interfaces/github.py]: i think you might also need to handle the 3rd case of a comment of length 1 (or just get rid of that case entirely)

File: inlineplz/interfaces/github.py [hunk 4]
```diff
@@ -92,3 +93,11 @@ def position(self, message):
                         if hunk_line.target_line_no == message.line_number:
                             return position + offset
                     offset += len(hunk) + 1
+
+    def clear_outdated_messages(self):
+        for comments in self.pull_request.review_comments():
```
Comment (guykisel) [inline inlineplz/interfaces/github.py]: `for comment`?

File: inlineplz/main.py [hunk 5]
```diff
@@ -27,6 +27,7 @@ def main():
     parser.add_argument('--zero-exit', action='store_true')
     parser.add_argument('--install', action='store_true')
     parser.add_argument('--max-comments', default=25, type=int, help='maximum comments to write')
+    parser.add_argument('--prefix', type=str)
```
Comment (guykisel) [inline inlineplz/main.py]: would it maybe be better to set the default prefix here? also maybe name this `--comment-prefix`? or is that too wordy?

File: inlineplz/interfaces/github.py [hunk 6]
```diff
@@ -92,3 +93,11 @@ def position(self, message):
...
+            if not comment.position and self.prefix in comment.body:
```
Comment (guykisel) [inline inlineplz/interfaces/github.py]: maybe only run this if `self.prefix` is truthy? (in case someone wants to disable prefixes entirely)

**Comment by midniteio** [comment]
Closing this PR in favor of opening PR in a branch in the original repo.
